lib/Dialect/FIRRTL/Transforms/ResolveTraces.cpp

```diff
@@ -75,35 +76,68 @@ struct ResolveTracesPass : public ResolveTracesBase<ResolveTracesPass> {
   /// runOnOperation.
   NLATable *nlaTable;
 
+  /// Global symbol index used for substitutions, e.g., "{{42}}".  This value is
+  /// the _next_ index that will be used.
+  unsigned symbolIdx = 0;
+
+  /// Map of symbol to symbol index.  This is used to reuse symbol
+  /// substitutions.
+  DenseMap<Attribute, unsigned> symbolMap;
+
+  /// Symbol substitutions for the JSON verbatim op.
+  SmallVector<Attribute> symbols;
+
+  /// Indicates exclusive access to updating symbolIdx, symbolMap, and symbols.
+  /// This is populated during runOnOperation.
+  llvm::sys::SmartMutex<true> *symbolMutex;
```

Comment:
This seems to make `symbols` and substituted indexes non-deterministic. Can we instead parallelly accumulate trace annotations and apply  `updateTargetImpl` sequentially?

Reply:
Updated to make this deterministic. The latest version of this PR changes the return type of the parallel region from `SmallVector<Annotation>` to `SmallVector<Annotation, AnnoPathValue>`. In the subsequent serial region each of these items is processed to generate the verbatim string targets with substitutions and a stable symbol order.